File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,vector<int>>mp;
        for(int i=0;i<nums.size();i++){
            mp[nums[i]].push_back(i);
        }
        vector<int>ans;

        for(int i=0;i<nums.size();i++){
            int remain = target-nums[i];

            if(mp[remain].size() > 0){
                int indexi=i,indexj;
                vector<int>v=mp[remain];
                int count=0;
                for(int j=0;j<v.size();j++){
                    if(i!=v[j]){
                        indexj=v[j];count=1;
                        ans.push_back(indexi),ans.push_back(indexj);
                        break;
                    }
                }
                if(count==1){break;}
            }
        }
        sort(ans.begin(),ans.end());
        return ans;
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        for index in range(0, len(nums)):
            if nums[index] not in dic:
                dic[nums[index]] = []   # create a new list
            dic[nums[index]].append(index)

        first = -1
        second = -1
        for index in range(0, len(nums)):
            remain = target-nums[index]
            if remain in dic:
                first=index
                array = dic[remain]
                for i in range(0, len(array)):
                    if (index != array[i]):
                        second=array[i]
                        break
            if second != -1:
                break

        return [first, second]
        